### PR TITLE
fitnesstrax: init at 0.1.0

### PR DESCRIPTION
**Pull request by savannidgerinel**
###### Motivation for this change

This adds a fitness tracking application that I am developing.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]:
```suggestion
{ pkgs
, fetchFromGitHub
,  stdenv
,  rustPlatform
, wrapGAppsHook
}:
```

callPackage is going to fill this in properly already

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]: please reference these formally (without pkgs). All of these except `wrapGAppsHook` should be `buildInputs` also.

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]:
```suggestion
  '';
```

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]: can you add yourself to `maintainers-list.nix`?

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]:
```suggestion
    description = "Privacy-first fitness tracking. Your health, your data, your machine";
```

the docs on meta attributes in nixpkgs say no ending period

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]:
```suggestion
  pame = "fitnesstrax";
```

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]: do you have a tag in your repo? then you could do `rev = version;`

**Review by worldofpeace — CHANGES_REQUESTED** [review]

**Comment by worldofpeace** [comment]
Why doesn't your makefile do what is done in `postInstall`. I could see if I could find a project that does exactly this as an example.

**Comment by savannidgerinel** [comment]
> Why doesn't your makefile do what is done in `postInstall`? I could see if I could find a project that does exactly this as an example.

I haven't seen makefiles in Rust projects before, but I've also not seen much in the way of Rust-based executables. Perhaps this is just a place where the ecosystem doesn't have sufficient known standards?

If you have an example, I would definitely follow it. Otherwise, I thought I had seen in my experiments that the shared/ folder just doesn't get copied into the destination.

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (savannidgerinel) [reply pkgs/applications/misc/fitnesstrax/default.nix]: Sweet. I did not know that was possible.

**Comment by worldofpeace** [comment]
> Why doesn't your makefile do what is done in `postInstall`? I could see if I could find a project that does exactly this as an example.

GLib provides https://gitlab.gnome.org/GNOME/glib/-/blob/master/m4macros/gsettings.m4, but you 'd need to use autotools. In that case I would really just recommend using meson.

**Comment by worldofpeace** [comment]
Oh, I thought you were using makefile somehow from looking at your project. Nevermind that then.

**Comment by worldofpeace** [comment]
I think you might want to delete this file from your repo https://github.com/luminescent-dreams/fitnesstrax/blob/sol/share/glib-2.0/schemas/gschemas.compiled. It's compiled but not meant to be distributed in that sense.

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (savannidgerinel) [reply pkgs/applications/misc/fitnesstrax/default.nix]: Done, plus added the other formal dependencies.

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (savannidgerinel) [reply pkgs/applications/misc/fitnesstrax/default.nix]: Split. only needed wrapGAppsHook in nativeBuildInputs, and the rest in work in buildInputs.

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (savannidgerinel) [reply pkgs/applications/misc/fitnesstrax/default.nix]: Tagged at fitnesstrax-0.1.0!

**Comment by savannidgerinel** [comment]
> I think you might want to delete this file from your repo https://github.com/luminescent-dreams/fitnesstrax/blob/sol/share/glib-2.0/schemas/gschemas.compiled. It's compiled but not meant to be distributed in that sense.

I'll do that in the next release. I recall needing it to get the derivation right, but it may have been an intermediate step while I was flailing.

**Comment by savannidgerinel** [comment]
Having addressed the changes, @worldofpeace and I get different sha256sums on the cargo file. I'm going to hold off until a few days and try stabilizing things on nixos-unstable, as suggested in IRC.

File: maintainers/maintainer-list.nix
Comment (worldofpeace) [inline maintainers/maintainer-list.nix]: I believe this has to be alphabetical?

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]:
```suggestion
    gdk-pixbuf
```

that's an old alias, it was renamed

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]: where do you use dlang bindings?

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]: :( sucks that you have to use the old version of pango

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix]:
```suggestion
    maintainers = [  maintainers.savanni ];
```

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (savannidgerinel) [reply pkgs/applications/misc/fitnesstrax/default.nix]: Yeah, doesn't appear that there's a new derivation. But since I don't use it directly, I don't so much notice.

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (savannidgerinel) [reply pkgs/applications/misc/fitnesstrax/default.nix]: Sorry, I don't understand the question. What do you mean?

File: maintainers/maintainer-list.nix
Comment (savannidgerinel) [reply maintainers/maintainer-list.nix]: Done. That wasn't entirely clear, but it looks like the end of the file has just gotten out of order.

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [reply pkgs/applications/misc/fitnesstrax/default.nix]: gtdk is `D binding and OO wrapper for GTK` https://dlang.org/. Isn't this all in rust?

File: pkgs/top-level/all-packages.nix
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]:
```suggestion
  fitnesstrax = callPackage ../applications/misc/fitnesstrax/default.nix { };
```

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (savannidgerinel) [reply pkgs/applications/misc/fitnesstrax/default.nix]: Going to guess that was a leftover of some early flailing as I was trying to chase down all of the dependencies, and that I forgot about it.

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix:22]:
```suggestion
    rev = "${pname}-${version}";
```

**Comment by worldofpeace** [comment]
I noticed something up with your deps, which I actually fixed for you https://github.com/NixOS/nixpkgs/pull/83336/commits/a9f12d5f5ab2dd733f3cf43270d87f7e121f8a68.
Only system libs you'd need would have been for `gtk-rs`.

**Comment by savannidgerinel** [comment]
@worldofpeace I am not sure how to proceed. I had no idea that you could make a commit to a PR from a remote repository, so I don't know how to integrate this into my PR.

And, I don't understand the error that grahamcofborg-eval is reporting.

What do you advise as my next step?

**Comment by worldofpeace** [comment]
@savannidgerinel Just pull from origin.

The error is a problem with ofborg that graham is fixing/maybe fixed.

Will try an eval
@GrahamcOfBorg build fitnesstrax

But generally, if you could build the package it should be fine.
(though the metadata could be messed up, but I don't think it is)

File: pkgs/applications/misc/fitnesstrax/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/fitnesstrax/default.nix:16]: you should switch the `name` attr to `pname` at a lower level they are concatenated like `${pname}-${version}` to make `name` But it's not available at an attribute level. Currently the derivation name is just `fitnesstrax`.

Comment (savannidgerinel) [reply pkgs/applications/misc/fitnesstrax/default.nix]: done!

**Comment by worldofpeace** [comment]
@savannidgerinel Squash merged, thanks :sparkles:

**Pull request merged**
